sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -1154,6 +1192,71 @@ private EventProcessorCheckpoint CreateCheckpointWithDefaultStartingPosition(str
             };
         }
 
+        /// <summary>
+        ///   Performs the tasks needed to validate basic configuration and permissions of the dependencies needed for
+        ///   the processor to function.
+        /// </summary>
+        ///
+        /// <param name="async">When <c>true</c>, the method will be executed asynchronously; otherwise, it will execute synchronously.</param>
+        /// <param name="containerClient">The <see cref="BlobContainerClient" /> to use for validating storage operations.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the start operation.</param>
+        ///
+        /// <exception cref="AggregateException">Any validation failures will result in an aggregate exception.</exception>
+        ///
+        private async Task ValidateStartupAsync(bool async,
+                                                BlobContainerClient containerClient,
+                                                CancellationToken cancellationToken = default)
+        {
+            var blobClient = containerClient.GetBlobClient(Guid.NewGuid().ToString("N"));
+
+            // Write an blob with metadata, simulating the approach used for checkpoint and ownership
+            // data creation.
+
+            try
+            {
+                using var blobContent = new MemoryStream(Array.Empty<byte>());
+                var blobMetadata = new Dictionary<string, string> {{ "name", blobClient.Name }};
```

Comment:
nit: Just in case the cleanup fails, perhaps give it a discoverable tag like "EventHubsValidateStartup"?